test/lensPath.js

```diff
@@ -0,0 +1,63 @@
+var R = require('..');
+var eq = require('./shared/eq');
+
+
+var testObj = {
+  a: {
+    b: 2,
+  },
+  d: 3
+};
+
+describe('lensPath', function() {
+  describe('view', function() {
+    it('focuses the specified object property', function() {
+      eq(R.view(R.lensPath(['d']), testObj), 3);
+      eq(R.view(R.lensPath(['a', 'b']), testObj), 2);
+      eq(R.view(R.lensPath([]), testObj), testObj);
+    });
+  });
+  describe('set', function() {
+    it('sets the value of the object property specified', function() {
+      eq(R.set(R.lensPath(['d']), 0, testObj), {a: {b: 2}, d: 0});
+      eq(R.set(R.lensPath(['a', 'b']), 0, testObj), {a: {b: 0}, d: 3});
+      eq(R.set(R.lensPath([]), 0, testObj), testObj);
```

Comment:
> Then again what does it actually mean to focus on an empty path, doesn't that mean we are not actually focusing anything? Not obvious to me that the empty path would yield the whole object.

Imagine standing in the entrance of a building. You are told to cross the foyer, ascend the stairs to the first floor, turn left, walk down the hallway and enter the second door on the right. You've been given a "path" to a "focus". Your view of the building is now focused on the room to which you were directed. Had the "path" been instructions to remain standing in the building's entrace, your view would be of the entire building. This is how I conceptualize `[]`.